default.nix

```diff
@@ -257,4 +257,92 @@ rec {
   shell = pkgs.mkShellNoCC {
     packages = [ ];
   };
+
+  demo-system =
+    module:
+    let
+      nixosSystem =
+        args:
+        import (sources.nixpkgs + "/nixos/lib/eval-config.nix") (
+          {
+            inherit lib;
+            system = null;
+          }
+          // args
+        );
+    in
+    nixosSystem {
+      system = "x86_64-linux";
+      modules = [
+        module
+        (sources.nixpkgs + "/nixos/modules/profiles/qemu-guest.nix")
+        (sources.nixpkgs + "/nixos/modules/virtualisation/qemu-vm.nix")
+        (
+          { config, ... }:
+          {
+            users.users.nixos = {
+              isNormalUser = true;
+              extraGroups = [ "wheel" ];
+              initialPassword = "nixos";
+            };
+
+            users.users.root = {
+              initialPassword = "root";
+            };
+
+            security.sudo.wheelNeedsPassword = false;
+
+            services.getty.autologinUser = "nixos";
+            services.getty.helpLine = ''
+
+              Welcome to Ngipkgs!
+            '';
+
+            virtualisation = {
+              memorySize = 4096;
+              cores = 4;
+              graphics = false;
+              diskImage = null;
```

Comment:
Having disk image is useful because it allows you to store some state between VM runs (data, shell history). Disadvantage is that it creates disk image file on your hard drive (qcow2 file).